Automattic/wp-calypso
# Domain Picker: refactor folder structure

#### Changes proposed in this Pull Request

* Refactor the structure of the Domain Picker `src` folder to allow for better modularity and tests granularity
* Refer to #51144 and to this PR's commit log for details list of changes

#### Testing instructions



These changes should only affect the internal structure of the Domain Picker package, and therefore shouldn't affect any user-facing behaviour

* Run `yarn && yarn start` and make sure that the project builds correctly
* Run `cd apps/editing-toolkit && yarn build` and make sure that the project builds correctly
* Run `cd packages/domain-picker && yarn test` and make sure that all tests pass (the `<button> cannot appear as a descendant of <button>.` error is expected)
* Smoke test Gutenboarding and Focused Launch



Closes #51144
